### PR TITLE
Make event view page buttons functional and improve layout

### LINKED ISSUE #749: Improve events/view route
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

screenshot from 2017-08-19 07-22-48

- Add link to preview button
- Add functionality to publish button
- Improve event details
   -  Use `sessionsByCategory` for no. of sessions
   -  Add links to redirect to event edit page if information is missing eg. location

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Makes event view page buttons functional other than the copy button.

Fixes #749

### CODE REVIEW

**Comment by niranjan94** [comment]
Deployment link: https://pr-754-open-event-frontend-fossasia.surge.sh

**Comment by CosmicCoder96** [comment]
@geekyd Done !

**Comment by niranjan94** [comment]
Deployment link: https://pr-754-open-event-frontend-fossasia.surge.sh

File: app/templates/components/events/view/overview/general-info.hbs [hunk 1]
```diff
@@ -32,17 +38,25 @@
...
         <td>
-          {{#each data.sessionTypes as |sessionType index|}}
-            {{if (not-eq index 0) ','}} {{sessionType.name}}
-          {{/each}}
+          {{#if data.sessionTypes}}
+            {{#each data.sessionTypes as |sessionType index|}}
+              {{if (not-eq index 0) ','}} {{sessionType.name}}
+            {{/each}}
+          {{else}}
+            {{t 'No Session Types present. '}} <a href="{{href-to 'events.view.edit.sessions-speakers'}}">{{t 'Click here to add'}}</a>
```
Comment (geekyd) [inline app/templates/components/events/view/overview/general-info.hbs]: This should be `No session type present`

**Comment by niranjan94** [comment]
Deployment link: https://pr-754-open-event-frontend-fossasia.surge.sh

**Pull request merged**